rules_gapic/java/java_gapic.bzl

```diff
@@ -175,6 +176,17 @@ def java_gapic_library(
         "@javax_annotation_javax_annotation_api//jar",
     ]
 
+    if transport == "rest":
```

Comment:
I'm trying to keep it backward-compatible with the older version of the rules, when no transport argument meant grpc. If the implementation of gapic generator does not support rest+grpc I don't think it makes much sense to support that in the rules. The current implementation is both backward compatible and the simplest I could come up with. Doing removal would mean that `rest+grpc` and `grpc+rest` "rest" only which is confusing and more complicated behavior.

Reply:
+1 to backwards-compatibility.

Reply:
Note, all this is essentially a throw away code, because the final destination of it will be in java microgenerator. Please lets not try to polish throwaway code or add features which most likely will never be fully implemented here (like the http+grpc option).

Reply:
It's a very simple change that increases clarity, IMO. My preference would be to add it. But as I said, I'm not going to block on this.